src/main/java/com/epam/learning/backendservices/springfoundation/dao/UserDAO.java

```diff
@@ -0,0 +1,35 @@
+package com.epam.learning.backendservices.springfoundation.dao;
+
+import com.epam.learning.backendservices.springfoundation.data.UserRepository;
+import com.epam.learning.backendservices.springfoundation.models.User;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.UUID;
+
+@Component
+public class UserDAO {

```

Comment:
1. This class does more look like a service rather than DAO. UserRepository is DAO. Rename both class and package to service.

```java
public interface UserService {
//...
}

public class UserServiceImpl implements UserService {
//...
}
```
2. Since you are working with database, add org.springframework.transaction.annotation.Transactional on the class level

Reply:
Done